ipld/retriever.go

```diff
@@ -172,27 +176,26 @@ func (rs *retrieverSession) request(ctx context.Context, q *quadrant) {
 	wg := &sync.WaitGroup{}
 	wg.Add(size)
 
-	log.Debugw("requesting quadrant", "x", q.x, "y", q.y, "size", size)
+	log.Debugw("requesting quadrant", "axis", q.source, "x", q.x, "y", q.y, "size", size)
 	for i, root := range q.roots {
 		go func(i int, root cid.Cid) {
 			defer wg.Done()
+			// get the root node
 			nd, err := rs.dag.Get(ctx, root)
 			if err != nil {
-				log.Errorw("getting root", "root", root.String(), "err", err)
 				return
 			}
-			// TODO(@Wondertan): GetLeaves should return everything it was able to request even on error,
-			// 	so that we fill as much data as possible
-			// get leaves of left or right subtree
-			nds, err := GetLeaves(ctx, rs.dag, nd.Links()[q.x].Cid, make([]format.Node, 0, size))
-			if err != nil {
-				log.Errorw("getting all the leaves", "root", root.String(), "err", err)
-				return
-			}
-			// fill leaves into the square
-			for j, nd := range nds {
-				rs.square[q.index(i, j)] = nd.RawData()[1+NamespaceSize:]
-			}
+			// and go get shares of left or the right side of the whole col/row axis
```

Comment:
You are really only getting shares of left or right side of the **row** axis, since you are using `nd.Links()[q.x].CID` correct?

Reply:
Yes and this was part of the previous PR. The left or right sides are identified by a quadrant we are downloading.

Reply:
Okay so can you change then the comment to say `go get shares of left or right side of the row axis` bc you're only using the row coordinate to determine this.

Reply:
Well, it is also about columns actually

Reply:
Check out code that creates quadrants

Reply:
newQuadrants and check for coordinates swapping in there

Reply:
ack